rio_tiler/models.py

```diff
@@ -421,18 +422,26 @@ def create_from_list(cls, data: Sequence["ImageData"]) -> "ImageData":
 
         """
         h, w = zip(*[(img.height, img.width) for img in data])
+
+        # Get cutline mask at highest resolution.
+        max_h, max_w = max(h), max(w)
+        cutline_mask = next(
+            img.cutline_mask
+            for img in data
+            if img.height == max_h and img.width == max_w
+        )
+
         if len(set(h)) > 1 or len(set(w)) > 1:
             warnings.warn(
                 "Cannot concatenate images with different size. Will resize using max width/heigh",
                 UserWarning,
             )
-            max_h, max_w = max(h), max(w)
             for img in data:
                 if img.height == max_h and img.width == max_w:
                     continue
                 arr = numpy.ma.MaskedArray(
                     resize_array(img.array.data, max_h, max_w),
-                    mask=resize_array(img.array.mask * 1, max_h, max_w).astype("bool"),
+                    mask=cutline_mask,
```

Comment:
You are absolutely right again! But that opens the problem that pixels might never get unmasked when upsampling. In the following example, the green pixel would always be masked at the coarse resolution, but at the higher resolution it would not be masked. So the finish early would not work again. The solution is to go back to what you already suggested earlier - setting always  `all_touched` to `True`. In that case the finer mask should always be included in the coarser mask. Will make change accordingly.

![image](https://user-images.githubusercontent.com/901647/234886166-781d78be-d527-4c26-89b8-3bfa7e3f7e44.png)